### PR TITLE
[sc-65908] Work around some schema improvements

### PR DESCRIPTION
**Pull request by tomascasas**
This one includes a few changes to support:
* Assets' paths
* Ad schema attached to the `'/v1/flight/{flightId}/creative'` path
* Job paths and expanding its schema

### CODE REVIEW

**Comment by honeycomb-cheesecake** [comment]
# Summary

I used a merger tool as my first test to resolve the entire management specification without references, and on main there are no hanging references (i.e. refs that couldn't be resolved), but with these changes there are a few. I'll comment where the issues are.

# Results

* Main [openapi-3-resolved-refs.main.txt](https://github.com/user-attachments/files/17990714/openapi-3-resolved-refs.main.txt)
* This PR [openapi-3-resolved-refs.txt](https://github.com/user-attachments/files/17990686/openapi-3-resolved-refs.txt)

File: management/schemas/job.yaml [hunk 1]
```diff
@@ -1,23 +1,135 @@
...
+          FlightId:
+            type: integer
+            format: int32
+          Overrides:
+            type: object
+            description: 'A map of ProductUri keys with Override values'
+            additionalProperties:
+              $ref: '#/schemas/Override'
 
-  JobOutput:
```
Comment (honeycomb-cheesecake) [inline management/schemas/job.yaml:16]: The removal of this schema definition means the reference in `management/schemas/flight.yaml` doesn't resolve anymore.

```
Jobs:
      type: array
      items:
        $ref: './job.yaml#/schemas/JobOutput'
```

File: management/asset.yaml [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+      security:
+        - ApiKeyAuth: []
+      responses:
+        200:
+          description: Successfully Deleted
+  '/v1/asset/{assetId}/upload':
+    parameters:
+      - name: assetId
+        in: path
+        description: Id of the Asset to delete
```
Comment (honeycomb-cheesecake) [inline management/asset.yaml:149]: Description for upload has a delete description.

File: management/asset.yaml [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+        required: true
+        schema:
+          type: integer
+          format: int32
+    post:
+      tags:
+        - assets
+      description: Upload an existing Asset Content
```
Comment (honeycomb-cheesecake) [inline management/asset.yaml:157]: Maybe "Upload content to and existing asset".

File: management/job.yaml [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+  - url: 'https://api.adzerk.net'
+tags:
+  - name: jobs
+    description: Create Jobs
+paths:
+  /v1/job/{jobId}:
+    get:
+      tags:
+        - jobs
+      description: List Jobs
```
Comment (honeycomb-cheesecake) [inline management/job.yaml:16]: Get Job

File: management/job.yaml [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+  title: Adzerk Management API - Advertisers
+  description: Advertiser related Adzerk Management API
+  version: '1.0'
+servers:
+  - url: 'https://api.adzerk.net'
+tags:
+  - name: jobs
+    description: Create Jobs
+paths:
+  /v1/job/{jobId}:
```
Comment (honeycomb-cheesecake) [inline management/job.yaml:12]: This is a get individual job endpoint.

File: management/job.yaml [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+    get:
+      tags:
+        - jobs
+      description: List Jobs
+      operationId: listJobs
```
Comment (honeycomb-cheesecake) [inline management/job.yaml:17]: getJob

File: management/job.yaml [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+            format: int32
+        - name: verbose
+          in: query
+          description: Whether to return TaskArgs and Output attributes in returned jobs
+          required: false
+          schema:
+            type: boolean
+      responses:
+        200:
+          description: The existing Job
```
Comment (honeycomb-cheesecake) [inline management/job.yaml:36]: The specified job.

File: management/openapi-3.yaml [hunk 8]
```diff
@@ -2034,13 +2038,259 @@ paths:
       responses:
         200:
           description: Successfully Deleted
+  /v1/job/{jobId}:
+    get:
+      tags:
+        - jobs
+      description: List Jobs
```
Comment (honeycomb-cheesecake) [inline management/openapi-3.yaml:2045]: Get job

File: management/openapi-3.yaml [hunk 9]
```diff
@@ -2034,13 +2038,259 @@ paths:
...
+      operationId: listJobs
```
Comment (honeycomb-cheesecake) [inline management/openapi-3.yaml:2046]: getJob

File: management/openapi-3.yaml [hunk 10]
```diff
@@ -2034,13 +2038,259 @@ paths:
...
+      security:
+        - ApiKeyAuth: []
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: './schemas/job.yaml#/schemas/JobInput'
+      responses:
+        200:
+          description: Advertiser Created
```
Comment (honeycomb-cheesecake) [inline management/openapi-3.yaml:2122]: Job created.

File: management/openapi-3.yaml [hunk 11]
```diff
@@ -2034,13 +2038,259 @@ paths:
...
+          description: Paged list of Assets
+          content:
+            application/json:
+              schema:
+                $ref: './schemas/asset.yaml#/schemas/AssetList'
+  '/v1/asset/{assetId}':
+    parameters:
+      - name: assetId
+        in: path
+        description: Id of the Asset to get
```
Comment (honeycomb-cheesecake) [inline management/openapi-3.yaml:2163]: Can be more general here in case more http verbs are added.

File: management/openapi-3.yaml [hunk 12]
```diff
@@ -2034,13 +2038,259 @@ paths:
...
+        200:
+          description: Paged list of Assets for Advertiser
+          content:
+            application/json:
+              schema:
+                $ref: './schemas/asset.yaml#/schemas/AssetList'
+    post:
+      tags:
+        - assets
+      description: Create Assets
```
Comment (honeycomb-cheesecake) [inline management/openapi-3.yaml:2225]: Create Asset

File: management/asset.yaml [hunk 13]
```diff
@@ -0,0 +1,178 @@
...
+        200:
+          description: Paged list of Assets for Advertiser
+          content:
+            application/json:
+              schema:
+                $ref: './schemas/asset.yaml#/schemas/AssetList'
+    post:
+      tags:
+        - assets
+      description: Create Assets
```
Comment (honeycomb-cheesecake) [inline management/asset.yaml:110]: Create Asset

File: management/openapi-3.yaml [hunk 14]
```diff
@@ -2034,13 +2038,259 @@ paths:
...
+          description: Asset Created
+          content:
+            application/json:
+              schema:
+                $ref: './schemas/asset.yaml#/schemas/Asset'
+  '/v1/asset/{assetId}/delete':
+    parameters:
+      - name: assetId
+        in: path
+        description: Id of the Asset to delete
```
Comment (honeycomb-cheesecake) [inline management/openapi-3.yaml:2245]: can be more general here to be applicable to other verbs that might be added.

File: management/asset.yaml [hunk 15]
```diff
@@ -0,0 +1,178 @@
...
+          description: Paged list of Assets
+          content:
+            application/json:
+              schema:
+                $ref: './schemas/asset.yaml#/schemas/AssetList'
+  '/v1/asset/{assetId}':
+    parameters:
+      - name: assetId
+        in: path
+        description: Id of the Asset to get
```
Comment (honeycomb-cheesecake) [inline management/asset.yaml:48]: Can be more general here.

File: management/openapi-3.yaml [hunk 16]
```diff
@@ -2034,13 +2038,259 @@ paths:
...
+      security:
+        - ApiKeyAuth: []
+      responses:
+        200:
+          description: Successfully Deleted
+  '/v1/asset/{assetId}/upload':
+    parameters:
+      - name: assetId
+        in: path
+        description: Id of the Asset to delete
```
Comment (honeycomb-cheesecake) [inline management/openapi-3.yaml:2264]: Delete description for upload, also can be more general here.

File: management/openapi-3.yaml [hunk 17]
```diff
@@ -2034,13 +2038,259 @@ paths:
...
+        required: true
+        schema:
+          type: integer
+          format: int32
+    post:
+      tags:
+        - assets
+      description: Upload an existing Asset Content
```
Comment (honeycomb-cheesecake) [inline management/openapi-3.yaml:2272]: Maybe "Upload content to and existing asset".

File: management/schemas/ad.yaml [hunk 18]
```diff
@@ -1,28 +1,13 @@
 schemas:
-
-  Ad:
+  BaseAd:
```
Comment (honeycomb-cheesecake) [inline management/schemas/ad.yaml:2]: Should this be `AdBase`?

File: management/schemas/job.yaml (hunk 1 above)
Comment (tomascasas) [reply management/schemas/job.yaml:16]: Thanks for flagging this one. That is now `Job`.

File: management/schemas/asset.yaml [hunk 19]
```diff
@@ -0,0 +1,108 @@
...
+      - type: object
+        properties:
+          ContentURL:
+            type: string
+            nullable: true
+            optional: true
+
+  Asset:
+    allOf:
+      - $ref: '#/schemas/AdBase'
```
[the remark this answers is not in the document]
Comment (tomascasas) [reply management/schemas/asset.yaml:26]: Thanks again. I now renamed `BaseAd` to `AdBase` and `BaseAsset` to `AssetBase` to match naming pattern across specs

File: management/asset.yaml (hunk 3 above)
Comment (tomascasas) [reply management/asset.yaml:157]: That's classic spanglish! Thank you!

File: management/job.yaml (hunk 5 above)
Comment (tomascasas) [reply management/job.yaml:12]: Was considering that my brain forgot porting from `job.yaml` to `openapi-3.yaml`. But this proves I duplicated and did not update `description` nor `operationId` for the single job getter.
Thank you for capturing both!

File: management/schemas/ad.yaml [hunk 20]
```diff
@@ -137,6 +125,34 @@ schemas:
         format: int32
         nullable: true
 
+  AdInput:
+    allOf:
+      - $ref: '#/schemas/AdBase'
```
[the remark this answers is not in the document]
Comment (tomascasas) [reply management/schemas/ad.yaml:130]: Thanks again. I now renamed `BaseAd` to `AdBase` and `BaseAsset` to `AssetBase` to match naming pattern across specs.
I had the intention of uniforming names to match other `[Entity]Base` but failed to do that properly.

File: management/asset.yaml [hunk 21]
```diff
@@ -0,0 +1,178 @@
...
+  '/v1/asset/{assetId}/upload':
+    parameters:
+      - name: assetId
+        in: path
+        description: Id of the Asset
+        required: true
+        schema:
+          type: integer
+          format: int32
+    post:
```
Comment (honeycomb-cheesecake) [inline management/asset.yaml:154]: This also has some required query parameters, you can see from this example path generated by the api below. Required query params are:

* timestamp: integer + required
* signature: string + required

```
"[HOST]/v1/asset/[assetId]/upload?timestamp=1234567&signature=aabbccddeeffgg"
```

Comment (tomascasas) [reply management/asset.yaml:154]: Thank you so much for capturing this one and walking me through the process to properly test all these.

**Review by honeycomb-cheesecake — APPROVED** [review]

File: management/schemas/campaign.yaml [hunk 22]
```diff
@@ -58,27 +58,29 @@ schemas:
...
+
   Campaign:
     allOf:
       - $ref: '#/schemas/CampaignBase'
       - type: object
-        required:
-          - Id
-          - StartDate
         properties:
+          required:
```
Comment (vkurup) [inline management/schemas/campaign.yaml:82]: Is this location correct? `required` previously used to be at the top level, but now it is underneath `properties`

**Review by vkurup — APPROVED** [review]

File: management/schemas/campaign.yaml (hunk 22 above)
Comment (tomascasas) [reply management/schemas/campaign.yaml]: Thanks for flagging this one!

**Pull request merged**